### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by bancer**

### CODE REVIEW

File: src/Traits/FactoryTrait.php [hunk 1]
```diff
@@ -50,7 +50,7 @@ public static function instance(DateTimeInterface $dt)
      * (new Chronos('Monday next week'))->fn()
      *
      * @param string $time The strtotime compatible string to parse
-     * @param \DateTimeZone|string|null $tz The DateTimeZone object or timezone name.
+     * @param \DateTimeInterface|\DateTimeZone|string|null| $tz The DateTimeZone object or timezone name.
```
Comment (garas) [inline src/Traits/FactoryTrait.php]: Last `|` looks wrong

File: src/Traits/FactoryTrait.php [hunk 2]
```diff
@@ -50,7 +50,7 @@ public static function instance(DateTimeInterface $dt)
      * (new Chronos('Monday next week'))->fn()
      *
      * @param string $time The strtotime compatible string to parse
-     * @param \DateTimeZone|string|null $tz The DateTimeZone object or timezone name.
+     * @param \DateTimeInterface|\DateTimeZone|string|null $tz The DateTimeZone object or timezone name.
```
Comment (ADmad) [inline src/Traits/FactoryTrait.php]: You probably meant to change type of first argument, not second.

**Comment by dereuromark** [comment]
We should try to get phpstan to level 5, this would help find some of those issues.

**Comment by bancer** [comment]
Scrutinizer also detects them.

**Comment by dereuromark** [comment]
We should stop 5.5 in travis matrix IMO. 5.6 at this point should be minimum.

**Pull request merged**